src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -309,14 +309,18 @@ static Solution UnifyLinkedDocumentContents(Solution oldSolution, Solution newSo
 
         static Solution UpdateAddedDocumentToExistingContentsInSolution(Solution solution, DocumentId addedDocumentId)
         {
-            var relatedDocumentIds = solution.GetRelatedDocumentIds(addedDocumentId);
-            foreach (var relatedDocumentId in relatedDocumentIds)
-            {
-                var relatedDocument = solution.GetRequiredDocument(relatedDocumentId);
-                return solution.WithDocumentContentsFrom(addedDocumentId, relatedDocument.DocumentState, forceEvenIfTreesWouldDiffer: false);
-            }
+            // Look for a related document we can create our contents from.  We only have to look for a single related
+            // doc as we'll be done once we update our contents to theirs.  Note: GetFirstRelatedDocumentId will also
+            // not search the project that addedDocumentId came from.  So this will help ensure we don't repeatedly add
+            // documents to a project, then look for related docs *within that project*, forcing the file-path map in it
+            // to be recreated for each document.
```

Comment:
this is a *nice* benefit, esp. given that we will often add a lot of docs at the same time to a solution.